### PR TITLE
Document theory in the md component.

### LINKED ISSUE #1078: Revise documentation to include complete theory descriptions
## Description

Expand the documentation of the box, MD integrators, pair potentials, bond potentials, and other classes with a mathematical description of the computation performed. Where needed, include figures to help explain.

## Motivation and context

HOOMD's current documentation makes a number of assumptions that readers understand certain terms (e.g. exclusion). Not all readers have the same background in simulation codes to understand these terms, or they may be familiar with codes that use these terms differently. Documenting these with mathematical equations will make the computations that HOOMD performs more explicit and understandable by readers. After this change, there will still be built in assumptions, but the assumed knowledge will be textbook material (e.g. potential energy, Hamiltonian).

We also need to make consistent use of symbols across the user documentation. For example, we currently use `V` for both potential energy and volume.

Related to #1027

### LINKED ISSUE #1229: Remove versionadded and versionchanged documentation directives
## Description

Remove the remaining `versionadded` and `versionchanged` directives in the sphinx documentation.

## Motivation and context

All APIs have changed in v3.0, this does not need to be explicitly documented. Starting with 3.1, we will begin adding these directives to describe the changes from 3.0.

### LINKED ISSUE #1262: Overlapping neighborlist exclusions cause errors in Coulomb/Ewald energies
## Description
I am mainly making this issue for purposes of documentation and because the result was unexpected.

I am running rigid body simulations of SPC/E water and I've found that my Coulomb.energy, Coulomb.additional_energy, and Ewald.energy differ depending on my choice of neighborlist exclusions _even when the exclusions logically cover the same particles._ For example in a rigid water model where each water molecule is a rigid body the exclusions: ["bond" "1-3", "body"], ["bond",  "1-3"], and ["body"] should cover the same particles; however, the resulting energies are different. In order to get agreement with the electrostatic energies of LAMMPS and Gromacs, I need to use the ["body"] or ["bond", "1-3"] exclusions. If I use the ["bond" "1-3", "body"] exclusion, there may be some double-counting.

### PR DESCRIPTION
**Pull request by joaander**
## Description

This is a full edit pass on the documentation of the `hoomd.md` package, with special attention to ensuring that the actions of operations are formally explained with mathematical notation (https://github.com/glotzerlab/hoomd-blue/issues/1078).

## Motivation and context

Ensure that the documentation is clear, consistent, accurate, and unambiguous.

Resolves #1078
Resolves #1262 
Resolves #1229

## How has this been tested?

I inspected the rendered documentation locally.

## Change log

```
Changed:

* [breaking] Renamed the ``hoomd.md.bond.Table`` energy parameter from ``V`` to ``U``.
* [breaking] Renamed the ``hoomd.md.pair.Table`` energy parameter from ``V`` to ``U``.
* [breaking] Renamed the ``hoomd.md.angle.Table`` energy parameter from ``V`` to ``U``.
* [breaking] Renamed the ``hoomd.md.dihedral.Table`` energy parameter from ``V`` to ``U``.
* [breaking] Renamed ``hoomd.md.nlist.Nlist`` to ``hoomd.md.nlist.NeighborList``.

Removed:

* The ``limit`` parameter from ``hoomd.md.methods.NVE``.
* The ``limit`` parameter from ``hoomd.md.methods.rattle.NVE``. 
* The ``diameter_shift`` parameter from `hoomd.md.nlist.NeighborList``.
* The ``max_diameter`` parameter from `hoomd.md.nlist.NeighborList``.
```

## Checklist:

- [X] I have reviewed the [**Contributor Guidelines**](https://github.com/glotzerlab/hoomd-blue/blob/master/CONTRIBUTING.md).
- [X] I agree with the terms of the [**HOOMD-blue Contributor Agreement**](https://github.com/glotzerlab/hoomd-blue/blob/master/ContributorAgreement.md).
- [X] My name is on the list of contributors (`sphinx-doc/credits.rst`) in the pull request source branch.

### CODE REVIEW

File: hoomd/md/pair/pair.py [hunk 1]
```diff
@@ -982,73 +806,56 @@ def __init__(self, nlist, default_r_cut=None):
...
-    functionality in your work.
-
-    .. math::
-        :nowrap:
-
-        \begin{eqnarray*}
-        F = F_{\mathrm{C}}(r) + F_{\mathrm{R,ij}}(r_{ij}) +
-            F_{\mathrm{D,ij}}(v_{ij}) \\
-        \end{eqnarray*}
+    `Buckingham` computes the `DPD` thermostat combined with the `LJ` pair force
```
Comment (b-butler) [inline hoomd/md/pair/pair.py:818]:
```suggestion
    `DPDDJ` computes the `DPD` thermostat combined with the `LJ` pair force
```

File: hoomd/md/pair/pair.py [hunk 2]
```diff
@@ -1565,7 +1321,7 @@ class ReactionField(Pair):
           :math:`[\mathrm{energy} \cdot \mathrm{length}]`
         * ``eps_rf`` (`float`, **required**) - :math:`\epsilon_{RF}`
           :math:`[\mathrm{dimensionless}]`
-        * ``use_charge`` (`boolean`, **optional**) - evaluate pair potential
+        * ``use_charge`` (`boolean`, **optional**) - evaluate pair force
```
Comment (b-butler) [inline hoomd/md/pair/pair.py:1324]:
```suggestion
        * ``use_charge`` (`bool`, **optional**) - evaluate pair force
```

**Review by b-butler — CHANGES_REQUESTED** [review]
This took a lot of effort. Great work. I have left comments for typos, fixes, and potential clarifications. For the most part I tried to use suggestions where applicable.

File: hoomd/md/integrate.py [hunk 3]
```diff
@@ -142,18 +142,16 @@ def _set_rigid(self, new_rigid):
 
 
 class Integrator(_DynamicIntegrator):
-    """Enables a variety of standard integration methods.
+    r"""Molecular dynamics time integration.
 
     Args:
-        dt (float): Integrator time step size :math:`[\\mathrm{time}]`.
+        dt (float): Integrator time step size :math:`[\mathrm{time}]`.
```
[the remark this answers is not in the document]
Comment (joaander) [reply hoomd/md/integrate.py:148]: I don't see anything wrong in the local build. Will check the rtd build.

File: sphinx-doc/module-md-nlist.rst [hunk 4]
```diff
@@ -6,19 +6,20 @@ md.nlist
...
+    Tree
 
 .. rubric:: Details
 
 .. automodule:: hoomd.md.nlist
     :synopsis: Neighbor list acceleration structures.
-    :members: NList, Cell, Stencil, Tree
+    :members: NeighborList, Cell, Stencil, Tree
     :no-inherited-members:
+    :show-inheritance:
```
[the remark this answers is not in the document]
Comment (joaander) [reply sphinx-doc/module-md-nlist.rst:25]: I prefer not to show empty bases. I used `autoclass` in other files, I just missed this one. Either way we list the classes, they need to be listed explicitly in the rst file - it makes no difference to me whether they are in the `:members:` list or as `autoclass` directives as needed with different options.

**Comment by joaander** [comment]
The changes to `readthedocs.yml` and `sphinx-doc/environment.yml` fix the readthedocs build failures currently affecting all branches. The switch to `mamba` cuts the time to build the docs down to 90 seconds (from 300+).

File: hoomd/md/integrate.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (joaander) [reply hoomd/md/integrate.py:148]: The readthedocs build looks the same. What do you see that isn't rendering correctly?

[the remark this answers is not in the document]
Comment (b-butler) [reply hoomd/md/integrate.py:148]: Hmm, it looked wrong on readthedocs when I first looked at it, but glad it/my eyes are fixed now.

**Comment by b-butler** [comment]
`mamba` for the win (I use it on all the machines I have access to).

**Review by b-butler — APPROVED** [review]
:+1:

**Pull request merged**